TestServerTracingExternal data race

Started appearing since the go agent update, no external reports of this so start with suspecting test setup issues.

```
11:19:38 WARNING: DATA RACE
11:19:38 Write at 0x00c0009ae910 by goroutine 321:
11:19:38   github.com/elastic/apm-server/vendor/go.elastic.co/apm.(*Tracer).loop.func1()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/vendor/go.elastic.co/apm/tracer.go:484 +0x3e
11:19:38   github.com/elastic/apm-server/vendor/go.elastic.co/apm.(*Tracer).loop()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/vendor/go.elastic.co/apm/tracer.go:561 +0x458e
11:19:38 
11:19:38 Previous read at 0x00c0009ae910 by goroutine 285:
11:19:38   github.com/elastic/apm-server/vendor/go.elastic.co/apm/module/apmhttp.(*handler).ServeHTTP()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/vendor/go.elastic.co/apm/tracer.go:328 +0x7b
11:19:38   net/http.serverHandler.ServeHTTP()
11:19:38       /var/lib/jenkins/.gvm/versions/go1.11.4.linux.amd64/src/net/http/server.go:2741 +0xc4
11:19:38   net/http.(*conn).serve()
11:19:38       /var/lib/jenkins/.gvm/versions/go1.11.4.linux.amd64/src/net/http/server.go:1847 +0x80a
11:19:38 
11:19:38 Goroutine 321 (running) created at:
11:19:38   github.com/elastic/apm-server/vendor/go.elastic.co/apm.newTracer()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/vendor/go.elastic.co/apm/tracer.go:267 +0x865
11:19:38   github.com/elastic/apm-server/vendor/go.elastic.co/apm.NewTracer()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/vendor/go.elastic.co/apm/tracer.go:234 +0xe4
11:19:38   github.com/elastic/apm-server/beater.initTracer()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/beater/beater.go:184 +0x16f
11:19:38   github.com/elastic/apm-server/beater.(*beater).Run()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/beater/beater.go:125 +0x120
11:19:38   github.com/elastic/apm-server/beater.setupBeater.func1()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/beater/beater_test.go:447 +0x48
11:19:38 
11:19:38 Goroutine 285 (running) created at:
11:19:38   net/http.(*Server).Serve()
11:19:38       /var/lib/jenkins/.gvm/versions/go1.11.4.linux.amd64/src/net/http/server.go:2851 +0x4c5
11:19:38   github.com/elastic/apm-server/beater.run()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/beater/server.go:92 +0x77c
11:19:38   github.com/elastic/apm-server/beater.(*beater).Run.func1()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/beater/beater.go:159 +0xc5
11:19:38   github.com/elastic/apm-server/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1()
11:19:38       /var/lib/jenkins/workspace/elastic+apm-server+master+beats-update/src/github.com/elastic/apm-server/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x64
11:19:38 ==================
11:19:38 --- FAIL: TestServerTracingExternal (0.59s)
11:19:38     testing.go:771: race detected during execution of test
11:19:38 FAIL
```

--- a/NOTICE.txt
+++ b/NOTICE.txt
@@ -2071,8 +2071,8 @@ OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 --------------------------------------------------------------------
 Dependency: go.elastic.co/apm
-Version: v1.1.1
-Revision: 69736506625fe7b07410587a5f993669f018b821
+Version: v1.1.2
+Revision: 9f89e36c6e534fd548810428cad1bd55cd57a5d2
 License type (autodetected): Apache-2.0
 ./vendor/go.elastic.co/apm/LICENSE:
 --------------------------------------------------------------------
@@ -2081,8 +2081,8 @@ Apache License 2.0
 
 --------------------------------------------------------------------
 Dependency: go.elastic.co/apm/internal/radix
-Version: v1.1.1
-Revision: 69736506625fe7b07410587a5f993669f018b821
+Version: v1.1.2
+Revision: 9f89e36c6e534fd548810428cad1bd55cd57a5d2
 License type (autodetected): MIT
 ./vendor/go.elastic.co/apm/internal/radix/LICENSE.goradix.txt:
 --------------------------------------------------------------------

--- a/changelogs/head.asciidoc
+++ b/changelogs/head.asciidoc
@@ -5,7 +5,7 @@
 ==== Added
 
 - Update Go version to 1.11.4 {pull}1700[1700].
-- Update Elastic APM Go agent to v1.1.1 {pull}1711[1711].
 - Remove formatting of `duration.us` to milliseconds in index pattern pull{1717}[1717].
+- Update Elastic APM Go agent to v1.1.2 {pull}1711[1711], {pull}1728[1728].
 
 https://github.com/elastic/apm-server/compare/v7.0.0-alpha2...master[View commits]

